pkg/feature/servicemesh/cleanup.go

```diff
@@ -0,0 +1,60 @@
+package servicemesh
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/gvr"
+)
+
+var log = ctrlLog.Log.WithName("features")
+
+func RemoveExtensionProvider(f *feature.Feature) error {
+	ossmAuthzProvider := fmt.Sprintf("%s-odh-auth-provider", f.Spec.AppNamespace)
+
+	mesh := f.Spec.ControlPlane
+
+	smcp, err := f.DynamicClient.Resource(gvr.SMCP).
+		Namespace(mesh.Namespace).
+		Get(context.TODO(), mesh.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	extensionProviders, found, err := unstructured.NestedSlice(smcp.Object, "spec", "techPreview", "meshConfig", "extensionProviders")
+	if err != nil {
+		return err
+	}
+	if !found {
+		log.Info("no extension providers found", "f", f.Name, "control-plane", mesh.Name, "namespace", mesh.Namespace)
```

Comment:
With #805 log becomes part of `Feature` struct. Depending on which PR comes first we will have to adjust it. It will make `"f", f.Name` redundant. And on that note, for now:

```suggestion
		log.Info("no extension providers found", "feature", f.Name, "control-plane", mesh.Name, "namespace", mesh.Namespace)
```

Reply:
Fixed